components/brave_extension/extension/brave_extension/manifest.json

```diff
@@ -77,7 +77,7 @@
       ]
     }
   ],
-  "permissions": [ "contentSettings", "management", "tabs", "storage", "webNavigation", "contextMenus", "cookies", "*://*/*", "chrome://favicon/*" ],
+  "permissions": [ "contentSettings", "settingsPrivate", "management", "tabs", "storage", "webNavigation", "contextMenus", "cookies", "*://*/*", "chrome://favicon/*" ],
```

Comment:
this concern actually applies to all `chrome.* APIs` that are specific to Brave, but what happens if someone is tricked into installing another extension that lists the `settingsPrivate` permission in its manifest?

does the attacker extension get access to `chrome.settingsPrivate` or is there something which checks that only the Brave extension (and other whitelisted extension IDs) can access this API?

Reply:
@diracdeltas `chrome.settingsPrivate` is from chromium and not a brave-addition. I was wondering the same thing. I assume either extensions are not allowed in the store, or that the permissions, which we have not modified, prevent external extensions from being allowed this API. The permissions are defined here: https://cs.chromium.org/chromium/src/chrome/common/extensions/api/_api_features.json?q=_api_features.json&sq=package:chromium&g=0&l=682

After testing side-loading a test extension which asks for settingsPrivate API into Brave stable and Chrome stable, the extension is not allowed access. It makes me wonder if Brave will also disable access for this internal extension too once we get an 'official' build from CI. We'll have to check, and if so modify _api_features.json as we do for some other APIs here: https://github.com/brave/brave-core/blob/48ad51c704278f574325b8e8f4904c819840dfaa/common/extensions/api/_api_features.json

![image](https://user-images.githubusercontent.com/741836/62248177-598a0b80-b39c-11e9-8afb-353d4a1854c8.png)

Reply:
oh it's actually here that I think I will need to modify https://cs.chromium.org/chromium/src/chrome/common/extensions/api/_permission_features.json?type=cs&q=permission+%22settingsPrivate%22&sq=package:chromium&g=0&l=678

Reply:
@petemill thanks for clarifying this is from chromium not brave. sgtm as long as it's whitelisted (or at least as long as we are no worse off than chrome in terms of allowing other extensions to access these APIs)

Reply:
@diracdeltas apologies for not specifying. Actually I did have to add an extra piece to whitelist the internal extension to this API: https://github.com/brave/brave-core/pull/2973/files#diff-fe4043ee5b80bbedb76102413057d29c

Reply:
added settingsPrivate API to Brave's overwriting _api_permissions.json and created https://github.com/brave/brave-browser/issues/5491 to stop overwriting upstream API security restrictions and instead merge in our own.